cli: TestRemoveDeadReplicas failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/d6adf24cae788d7cd967feadae8e9c0388ce5273

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestRemoveDeadReplicas PKG=github.com/cockroachdb/cockroach/pkg/cli TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1079976&tab=buildLog

```
E190104 06:13:32.036907 3439 storage/replica.go:1514  [n1,s1,r6/4:/Table/{SystemCon…-11}] lease repl=(n3,s3):2 seq=2 start=1546582411.279150002,0 epo=1 pro=1546582411.279152482,0 owned by replica (n3,s3):2 that no longer exists
W190104 06:13:32.041459 1806 server/status/runtime.go:294  [n?] Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
W190104 06:13:32.049362 1806 gossip/gossip.go:1499  [n?] no incoming or outgoing connections
W190104 06:13:32.078070 1806 server/status/recorder.go:389  [n4] unable to look up descriptor for n2
W190104 06:13:32.078096 1806 server/status/recorder.go:389  [n4] unable to look up descriptor for n3
W190104 06:13:32.078993 3609 server/status/recorder.go:389  [n4,summaries] unable to look up descriptor for n2
W190104 06:13:32.079024 3609 server/status/recorder.go:389  [n4,summaries] unable to look up descriptor for n3
E190104 06:13:32.100186 3397 storage/replica.go:1514  [n1,s1,r6/4:/Table/{SystemCon…-11}] lease repl=(n3,s3):2 seq=2 start=1546582411.279150002,0 epo=1 pro=1546582411.279152482,0 owned by replica (n3,s3):2 that no longer exists
E190104 06:13:32.100547 3439 storage/replica.go:1514  [n1,s1,r6/4:/Table/{SystemCon…-11}] lease repl=(n3,s3):2 seq=2 start=1546582411.279150002,0 epo=1 pro=1546582411.279152482,0 owned by replica (n3,s3):2 that no longer exists
E190104 06:13:32.100744 3416 storage/replica.go:1514  [n1,s1,r6/4:/Table/{SystemCon…-11}] lease repl=(n3,s3):2 seq=2 start=1546582411.279150002,0 epo=1 pro=1546582411.279152482,0 owned by replica (n3,s3):2 that no longer exists
E190104 06:13:32.102650 3504 storage/replica.go:1514  [n1,s1,r6/4:/Table/{SystemCon…-11}] lease repl=(n3,s3):2 seq=2 start=1546582411.279150002,0 epo=1 pro=1546582411.279152482,0 owned by replica (n3,s3):2 that no longer exists
E190104 06:13:32.103090 3408 storage/replica.go:1514  [n1,s1,r6/4:/Table/{SystemCon…-11}] lease repl=(n3,s3):2 seq=2 start=1546582411.279150002,0 epo=1 pro=1546582411.279152482,0 owned by replica (n3,s3):2 that no longer exists
E190104 06:13:32.103286 3669 storage/replica.go:1514  [n1,s1,r6/4:/Table/{SystemCon…-11}] lease repl=(n3,s3):2 seq=2 start=1546582411.279150002,0 epo=1 pro=1546582411.279152482,0 owned by replica (n3,s3):2 that no longer exists
W190104 06:13:32.107810 1806 server/status/runtime.go:294  [n?] Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
W190104 06:13:32.116802 1806 gossip/gossip.go:1499  [n?] no incoming or outgoing connections
E190104 06:13:32.130185 3354 storage/replica.go:1514  [n1,s1,r6/4:/Table/{SystemCon…-11}] lease repl=(n3,s3):2 seq=2 start=1546582411.279150002,0 epo=1 pro=1546582411.279152482,0 owned by replica (n3,s3):2 that no longer exists
W190104 06:13:32.130261 3354 storage/store.go:1466  [n1,s1,r6/4:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):4 not lease holder; lease holder unknown
W190104 06:13:32.141059 1806 server/status/recorder.go:389  [n5] unable to look up descriptor for n3
W190104 06:13:32.141088 1806 server/status/recorder.go:389  [n5] unable to look up descriptor for n2
W190104 06:13:32.141604 3822 server/status/recorder.go:389  [n5,summaries] unable to look up descriptor for n3
W190104 06:13:32.141712 3822 server/status/recorder.go:389  [n5,summaries] unable to look up descriptor for n2
E190104 06:13:32.146255 3678 storage/replica.go:1514  [n1,s1,r6/4:/Table/{SystemCon…-11}] lease repl=(n3,s3):2 seq=2 start=1546582411.279150002,0 epo=1 pro=1546582411.279152482,0 owned by replica (n3,s3):2 that no longer exists
E190104 06:13:32.146874 3677 storage/replica.go:1514  [n1,s1,r6/4:/Table/{SystemCon…-11}] lease repl=(n3,s3):2 seq=2 start=1546582411.279150002,0 epo=1 pro=1546582411.279152482,0 owned by replica (n3,s3):2 that no longer exists
--- FAIL: TestRemoveDeadReplicas (1.96s)
	testcluster.go:172: system config not available in gossip

```

SHA: https://github.com/cockroachdb/cockroach/commits/d6adf24cae788d7cd967feadae8e9c0388ce5273

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestRemoveDeadReplicas PKG=github.com/cockroachdb/cockroach/pkg/cli TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1080556&tab=buildLog

```
E190104 07:17:26.877985 3717 storage/replica.go:1514  [n1,s1,r6/4:/Table/{SystemCon…-11}] lease repl=(n2,s2):3 seq=2 start=1546586245.077613989,0 epo=1 pro=1546586245.077628467,0 owned by replica (n2,s2):3 that no longer exists
E190104 07:17:26.894907 3718 storage/replica.go:1514  [n1,s1,r6/4:/Table/{SystemCon…-11}] lease repl=(n2,s2):3 seq=2 start=1546586245.077613989,0 epo=1 pro=1546586245.077628467,0 owned by replica (n2,s2):3 that no longer exists
W190104 07:17:26.918374 3613 gossip/client.go:122  [n1] failed to start gossip client to 127.0.0.1:40131: initial connection heartbeat failed: rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:40131: connect: connection refused"
E190104 07:17:26.926475 3543 storage/replica.go:1514  [n1,s1,r6/4:/Table/{SystemCon…-11}] lease repl=(n2,s2):3 seq=2 start=1546586245.077613989,0 epo=1 pro=1546586245.077628467,0 owned by replica (n2,s2):3 that no longer exists
E190104 07:17:26.953055 3564 storage/replica.go:1514  [n1,s1,r6/4:/Table/{SystemCon…-11}] lease repl=(n2,s2):3 seq=2 start=1546586245.077613989,0 epo=1 pro=1546586245.077628467,0 owned by replica (n2,s2):3 that no longer exists
E190104 07:17:26.980401 3806 storage/replica.go:1514  [n1,s1,r6/4:/Table/{SystemCon…-11}] lease repl=(n2,s2):3 seq=2 start=1546586245.077613989,0 epo=1 pro=1546586245.077628467,0 owned by replica (n2,s2):3 that no longer exists
E190104 07:17:26.984402 3544 storage/replica.go:1514  [n1,s1,r6/4:/Table/{SystemCon…-11}] lease repl=(n2,s2):3 seq=2 start=1546586245.077613989,0 epo=1 pro=1546586245.077628467,0 owned by replica (n2,s2):3 that no longer exists
E190104 07:17:26.985696 3808 storage/replica.go:1514  [n1,s1,r6/4:/Table/{SystemCon…-11}] lease repl=(n2,s2):3 seq=2 start=1546586245.077613989,0 epo=1 pro=1546586245.077628467,0 owned by replica (n2,s2):3 that no longer exists
E190104 07:17:26.994796 3807 storage/replica.go:1514  [n1,s1,r6/4:/Table/{SystemCon…-11}] lease repl=(n2,s2):3 seq=2 start=1546586245.077613989,0 epo=1 pro=1546586245.077628467,0 owned by replica (n2,s2):3 that no longer exists
W190104 07:17:27.013816 1800 server/status/runtime.go:294  [n?] Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
W190104 07:17:27.083678 1800 gossip/gossip.go:1499  [n?] no incoming or outgoing connections
E190104 07:17:27.118397 3857 storage/replica.go:1514  [n1,s1,r6/4:/Table/{SystemCon…-11}] lease repl=(n2,s2):3 seq=2 start=1546586245.077613989,0 epo=1 pro=1546586245.077628467,0 owned by replica (n2,s2):3 that no longer exists
E190104 07:17:27.119393 3858 storage/replica.go:1514  [n1,s1,r6/4:/Table/{SystemCon…-11}] lease repl=(n2,s2):3 seq=2 start=1546586245.077613989,0 epo=1 pro=1546586245.077628467,0 owned by replica (n2,s2):3 that no longer exists
E190104 07:17:27.120341 3939 storage/replica.go:1514  [n1,s1,r6/4:/Table/{SystemCon…-11}] lease repl=(n2,s2):3 seq=2 start=1546586245.077613989,0 epo=1 pro=1546586245.077628467,0 owned by replica (n2,s2):3 that no longer exists
E190104 07:17:27.146306 3498 storage/replica.go:1514  [n1,s1,r6/4:/Table/{SystemCon…-11}] lease repl=(n2,s2):3 seq=2 start=1546586245.077613989,0 epo=1 pro=1546586245.077628467,0 owned by replica (n2,s2):3 that no longer exists
W190104 07:17:27.146635 3498 storage/store.go:1466  [n1,s1,r6/4:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):4 not lease holder; lease holder unknown
W190104 07:17:27.238956 1800 server/status/recorder.go:389  [n5] unable to look up descriptor for n3
W190104 07:17:27.239196 1800 server/status/recorder.go:389  [n5] unable to look up descriptor for n2
W190104 07:17:27.244083 3967 server/status/recorder.go:389  [n5,summaries] unable to look up descriptor for n3
W190104 07:17:27.246184 3967 server/status/recorder.go:389  [n5,summaries] unable to look up descriptor for n2
--- FAIL: TestRemoveDeadReplicas (7.31s)
	testcluster.go:172: system config not available in gossip

```

--- a/pkg/cli/debug_test.go
+++ b/pkg/cli/debug_test.go
@@ -30,6 +30,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/keys"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/server"
+	"github.com/cockroachdb/cockroach/pkg/server/serverpb"
 	"github.com/cockroachdb/cockroach/pkg/storage/engine"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
@@ -132,7 +133,6 @@ func TestOpenReadOnlyStore(t *testing.T) {
 
 func TestRemoveDeadReplicas(t *testing.T) {
 	defer leaktest.AfterTest(t)()
-	t.Skipf("33495")
 	ctx := context.Background()
 
 	baseDir, dirCleanupFn := testutils.TempDir(t)
@@ -236,10 +236,43 @@ func TestRemoveDeadReplicas(t *testing.T) {
 
 	// Now that the data is salvaged, we can restart the cluster. The
 	// nodes with the in-memory stores will be assigned new node IDs 4
-	// and 5. StartTestCluster will even wait for all the ranges to be
-	// replicated to the new nodes.
+	// and 5.
+	//
+	// This activates the adaptive zone config feature (using 5x
+	// replication for clusters of 5 nodes or more), so we must
+	// decommission nodes 2 or 3 for WaitForFullReplication to complete.
+	// (note that the cluster is working even when it doesn't consider
+	// itself fully replicated - that's what allows the decommissioning
+	// to succeed. We're just waiting for full replication so that we
+	// can validate that ranges were moved from {1,2,3} to {1,4,5}).
+	//
+	// Set replication mode to manual so that TestCluster doesn't call
+	// WaitForFullReplication before we've decommissioned the nodes.
+	clusterArgs.ReplicationMode = base.ReplicationManual
 	tc := testcluster.StartTestCluster(t, 3, clusterArgs)
 	defer tc.Stopper().Stop(ctx)
+
+	grpcConn, err := tc.Server(0).RPCContext().GRPCDial(tc.Server(0).ServingAddr()).Connect(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+	adminClient := serverpb.NewAdminClient(grpcConn)
+
+	if err := runDecommissionNodeImpl(
+		ctx, adminClient, nodeDecommissionWaitNone, []string{"2", "3"},
+	); err != nil {
+		t.Fatal(err)
+	}
+
+	store, err := tc.Servers[0].Stores().GetStore(1)
+	if err != nil {
+		t.Fatal(err)
+	}
+	store.SetReplicateQueueActive(true)
+	if err := tc.WaitForFullReplication(); err != nil {
+		t.Fatal(err)
+	}
+
 	s := sqlutils.MakeSQLRunner(tc.Conns[0])
 	row := s.QueryRow(t, "select replicas from [show experimental_ranges from table system.namespace] limit 1")
 	var replicaStr string

--- a/pkg/storage/store.go
+++ b/pkg/storage/store.go
@@ -885,7 +885,7 @@ func NewStore(cfg StoreConfig, eng engine.Engine, nodeDesc *roachpb.NodeDescript
 		s.setReplicaGCQueueActive(false)
 	}
 	if cfg.TestingKnobs.DisableReplicateQueue {
-		s.setReplicateQueueActive(false)
+		s.SetReplicateQueueActive(false)
 	}
 	if cfg.TestingKnobs.DisableSplitQueue {
 		s.setSplitQueueActive(false)
@@ -4514,7 +4514,10 @@ func (s *Store) setRaftLogQueueActive(active bool) {
 func (s *Store) setReplicaGCQueueActive(active bool) {
 	s.replicaGCQueue.SetDisabled(!active)
 }
-func (s *Store) setReplicateQueueActive(active bool) {
+
+// SetReplicateQueueActive controls the replication queue. Only
+// intended for tests.
+func (s *Store) SetReplicateQueueActive(active bool) {
 	s.replicateQueue.SetDisabled(!active)
 }
 func (s *Store) setSplitQueueActive(active bool) {

